### PR TITLE
CI: Update teleport version to 15.2.0

### PR DESCRIPTION
**Pull request by marcoandredinis**
Updates Teleport binary used in tests to 15.2.0

This new version of teleport has the following relevant changes:

### Teleport process logging uses `log/slog`
For terraform tests we start a teleport binary and parse its output to understand when the Auth/Proxy service started and what ports are they listening on.
We used a regex for that, but teleport migrated to `log/slog` and the regex no longer works.
Migration PR: https://github.com/gravitational/teleport/pull/38551
We had to fix the regex for integration tests in teleport: https://github.com/gravitational/teleport/pull/39315
Terraform Tests also use that library, so after the regex changed, we must upgrade Teleport CI version to get the new log format.

### Teleport API: `GetClusterNetworkingConfig` and `GetSessionRecordingConfig` never return a nil

When developing the `ClusterMaitenanceConfig` we had to include a nil check, because if it was never configured, `GetClusterMaintenanceConfig` would return a nil object.
This nil check was added to all SingleResource resources.

For `ClusterNetworkingConfig` and `SessionRecordingConfig`, the `Get` operation never returns a nil resource and `staticcheck` linter was yelling about it.
So, we had to create a new flag to ensure we only nil-checked the resources that can actually return a nil value.

### Teleport Resource Metadata
It is no longer recommended to use the `<Resource>.Metadata.ID` to check for cached responses.
We are now using the revision field.

During this change we also detected a miss-usage of an `error` variable and fixed that (could lead to a panic).

LoginRules didn't have the `Revision` field, so we added it here: https://github.com/gravitational/teleport.e/pull/3821

Unfortunately, that PR didn't merge in time for 15.2.0.
However, that's ok because LoginRules are not cached.
So, instead of waiting for a new release (15.2.1), we just removed the cache check.

### CODE REVIEW

File: terraform/Makefile [hunk 1]
```diff
@@ -103,7 +103,7 @@ release: build
 endif
 	tar -C $(BUILDDIR) -czf $(RELEASE).tar.gz .
 
-TERRAFORM_EXISTS := $(shell terraform -version 2>/dev/null | grep 'Terraform v1.5')
+TERRAFORM_EXISTS := $(shell terraform -version 2>/dev/null | grep 'Terraform v1.')
```
Comment (hugoShaka) [inline terraform/Makefile:106]: Is this change intentional? Do we want to force a specific TF version or we're OK with everything 1.x?

**Review by hugoShaka — APPROVED** [review]

File: terraform/Makefile (hunk 1 above)
Comment (marcoandredinis) [reply terraform/Makefile:106]: Yes, but I can add it into another PR if you think it's better

I think requiring anything in the `1.x` range is fine, and I was getting annoyed with having to upgrade multiple things

File: go.mod [hunk 2]
```diff
@@ -364,8 +364,8 @@ replace (
 	github.com/alecthomas/kingpin/v2 => github.com/gravitational/kingpin/v2 v2.1.11-0.20230515143221-4ec6b70ecd33
 	github.com/coreos/go-oidc => github.com/gravitational/go-oidc v0.1.1
 	github.com/gogo/protobuf => github.com/gravitational/protobuf v1.3.2-0.20201123192827-2b9fcfaffcbf
-	github.com/gravitational/teleport => github.com/gravitational/teleport v0.0.0-20240327020503-fc3b2b31dec6 // ref: tags/v15.1.10
-	github.com/gravitational/teleport/api => github.com/gravitational/teleport/api v0.0.0-20240327020503-fc3b2b31dec6 // ref: tags/v15.1.10
+	github.com/gravitational/teleport => github.com/gravitational/teleport v0.0.0-20240329210410-bb8bd77625f4 // ref: tags/v15.2.0
+	github.com/gravitational/teleport/api => github.com/gravitational/teleport/api v0.0.0-20240329210410-bb8bd77625f4 // ref: tags/v15.2.0
```
Comment (marcoandredinis) [inline go.mod:368]: @hugoShaka What do you think about this?
I'm upgrading to 15.2.0 here so we get the new regex used in integration tests
But, usually, only release PRs change this

Comment (hugoShaka) [reply go.mod:368]: I do this all the time to make the tests pass :)

The release PR will overwrite this anyway.

**Comment by marcoandredinis** [comment]
@r0mant Can you please take a look? I need a code owners review

**Review by r0mant — APPROVED** [review]
